### ISSUE TITLE
Can't run "debug locally" if port 5858 is already taken

### ISSUE THREAD

**Issue by anton-107** [aws/aws-toolkit-vscode#954]
**Describe the bug**
When local port 5858 is taken by another process and I hit "Debug locally" I get an error from docker saying: 
```
docker.errors.APIError: 500 Server Error: Internal Server Error ("driver failed programming external connectivity on endpoint [redacted] ([redacted]): Bind for 0.0.0.0:5858 failed: port is already allocated")
```

**To Reproduce**
0. Have another process running on port 5858
1. Go to a SAM function
2. Hit debug locally
3. See an error message in the output

**Expected behavior**
Pick a free port for debugging purposes instead of always picking the same one

**Comment by awschristou** [comment]
Thank you for filing the issue. We don't have a workaround at this time. Our current thought is to have the toolkit check for an available port each time a local debug session is started, and use that.

**Comment by GithubGoldMiner** [comment]
Hi @awschristou, I previously had the same issue as described above. I just checked `src/shared/utilities/vsCodeUtils.ts` and found that the debug port # is actually hard coded as 5858.
One potential solution I am thinking is to have a list of port numbers, for example, [5858 ... 5868] and iterate through that list using `net.createConnection()` to check if port is available. If it's available then return that port # for users to use. I think this might be better to just return 500 if that port is occupied.

**Comment by bryceitoc9** [comment]
PR: https://github.com/aws/aws-toolkit-vscode/pull/959/files

**Comment by GithubGoldMiner** [comment]
@bryceitoc9 A way much better solution by adding `portfinder`. Thank you for solving this issue!

**Comment by awschristou** [comment]
This change has been released with v1.8.0 of the toolkit
